examples/dielectric.jl

```diff
@@ -41,20 +43,23 @@ function arnoldi(f, x0, howmany; tol=1e-4, maxiter=30)
         AVr = AV*ev
         R = AVr - Vr * Diagonal(ew)
 
-        # Select `howmany` smallest and largest eigenpairs
         N = size(V, 2)
-        inds = unique(append!(collect(1:min(N, howmany)), max(1, N-howmany):N))
+        normr = [norm(r) for r in eachcol(R)]
 
-        normr = [norm(r) for r in eachcol(R[:, inds])]
         println("#--- $N ---#")
         println("idcs      evals     residnorms")
+        inds = unique(append!(collect(1:min(N, n_print)), max(1, N-n_print):N))
+        for i in inds
+            @printf "% 3i  %10.6g  %10.6g\n" i real(ew[i]) normr[i]
+        end
         any(imag.(ew[inds]) .> 1e-5) && println("Warn: Suppressed imaginary part.")
-        display(real.([inds ew[inds] normr]))
         println()
-        if (N ≥ howmany && maximum(normr) < tol) || (N ≥ maxiter)
-            return ew[inds], Vr[:, inds], AVr[:, inds]
+
+        is_converged = (N ≥ howmany && all(normr[1:howmany] .< tol)
+                        && all(normr[end-howmany:end] .< tol))
+        if is_converged || (N ≥ maxiter)
+            return (λ=ew, X=Vr, AX=AVr, residual_norms=normr)
         end
     end
 end
-howmany = 5
-arnoldi(eps_fun, vec(randn(size(scfres.ρ.real))), howmany)
+arnoldi(eps_fun, vec(randn(size(scfres.ρ.real))); howmany=5)
```

Comment:
We should replace that by krylovkit's native implementation anyway now that they have the "greedy" option. @gkemlin is doing that, we'll just merge later